### PR TITLE
[Master] Reduce the complexity of HTTP BBEs

### PR DESCRIPTION
**Pull request by chamil321**

### CODE REVIEW

**Comment by ayeshLK** [comment]
In `examples/http-100-continue/http_100_continue.bal`:
- Shall we remove unwanted parenthesis in `if` conditions ?
```ballerina
if (request.expects100Continue()) {
    ...
}
```
- Shall we use mime-constants for media-types ?
```ballerina
if mediaType.toLowerAscii() == "text/plain" {
    ...
}
```

File: examples/dynamic-listener/dynamic_listener.bal [hunk 1]
```diff
@@ -1,35 +1,24 @@
...
-        var respondResult = caller->respond("Hello, World!");
-        if respondResult is error {
-            log:printError("Error occurred when responding.", 'error = respondResult);
-        }
+        return "Hello, World!";
     }
 
     // The resource function that will shutdown the server.
-    resource function get shutDownServer(http:Caller caller, http:Request req) {
+    resource function get shutdown(http:Caller caller) returns error? {
```
Comment (manuranga) [inline examples/dynamic-listener/dynamic_listener.bal:14]: Should this be `get`, how about `post`?

File: examples/http-path-param/http_path_param.bal [hunk 2]
```diff
@@ -4,11 +4,7 @@ service /company on new http:Listener(9090) {
 
     // The path param is defined as a part of the resource path along with the type and it is extracted from the
     // request URI.
-    resource function get empId/[int id]() returns json {
-        return {empId: id};
-    }
-
-    resource function get empName/[string first]/[string last]() returns json {
-        return {firstName: first, lastName: last};
+    resource function get employee/[string id]() returns string {
```
Comment (sameerajayasoma) [inline examples/http-path-param/http_path_param.bal:7]: Returning string doesn't sound like a good example here?

File: examples/http-client-endpoint/http_client_endpoint.bal [hunk 3]
```diff
@@ -3,26 +3,17 @@ import ballerina/io;
...
     
     // Sends a `GET` request to the specified endpoint.
     io:println("GET request:");
-    json resp = check clientEndpoint->get("/get?test=123");
-    io:println(resp.toJsonString());
-
-    // The `get()`, `head()`, and `options()` have the optional headers parameter to send out headers,
-    io:println("\nGET request with Headers:");
-    resp = check clientEndpoint->get("/get", {"Sample-Name": "http-client-connector"});
+    json resp = check clientEndpoint->/get.get(test = 123);
```
Comment (sameerajayasoma) [inline examples/http-client-endpoint/http_client_endpoint.bal:10]: This is confusing as the first HTTP client example.

File: examples/http-client-data-binding/http_client_data_binding.bal [hunk 4]
```diff
@@ -1,66 +1,22 @@
...
-        string result = check backendClient->get("/backend/string");
-        log:printInfo("String payload: " + result);
-
-        // A `record` and `record[]` are also possible types for data binding.
-        Person person = check backendClient->get("/backend/person");
-        log:printInfo("Person name: " + person.name);
+service / on new http:Listener(9090) {
+    resource function get person() returns Person|error {
+        // Binding the payload to a `record`` type. The contextually expected type is inferred from the LHS variable type.
+        Person person = check backendClient->/backend/data;
```
Comment (sameerajayasoma) [inline examples/http-client-data-binding/http_client_data_binding.bal:13]: Let's use a meaningful client name here.

File: examples/http-client-endpoint/http_client_endpoint.bal [hunk 5]
```diff
@@ -3,26 +3,17 @@ import ballerina/io;
 
 public function main() returns error? {
```
Comment (sameerajayasoma) [inline examples/http-client-endpoint/http_client_endpoint.bal:4]: Shouldn't we talk about both `fooClient->get("/foos")` and `fooClient->/ foos`?

**Review by sameerajayasoma — CHANGES_REQUESTED** [review]

**Pull request closed without merging**